### PR TITLE
Multistore - Orders > Delivery Slips - Add checkboxes

### LINKED ISSUE #19383: Multistore - Orders > Delivery Slips - Add checkboxes
As a merchant, I want to be able to select one or many fields / toggles when I am in a specific shop context or in a group context, so I can apply a different configuration for the selected store or group and see which parameters have different value(s) for the selected store or group.

[Specs checkboxes](https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/multistoregeneralspecs.md#checkboxes)

### PR DESCRIPTION
**Pull request by nenes25**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Add multi store new behaviour on Delivery Slip Page , this PR need the implementation of https://github.com/PrestaShop/PrestaShop/pull/22139 to work properly
| Type?             | new feature 
| Category?         |  BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #19383.
| How to test?      | Multistore - Checkbox should be visible on field slip_prefix on page Orders > Delivery Slips
| Possible impacts? | Change the way the data are saved with multistore

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25957)

### CODE REVIEW

**Review by matthieu-rolland — COMMENTED** [review]
Thank you for your contribution @nenes25 !

Seems like cs fixer is not happy, you can fix it easily by executing this command from the root of your project:

`./vendor/friendsofphp/php-cs-fixer/php-cs-fixer fix src/PrestaShopBundle/Form/Admin/Sell/Order/Delivery/SlipOptionsType.php`

And then commit the modification 👍 

Also, I'm currently working on a PR that will add an example of unit tests for configuration classes, I hope it will help, hopefully it's ready tomorrow

**Comment by nenes25** [comment]
hi @matthieu-rolland ,
Thanks for the feedback i've fixed the Php Cs fixer issue.
I will update this PR with unit test once you'll release it :smile:

**Comment by PierreRambaud** [comment]
Hi @nenes25 
Are you planning to continue the work you started?

**Comment by nenes25** [comment]
@PierreRambaud  yes i'll try to finish it this soon ;)

File: src/PrestaShopBundle/Form/Admin/Sell/Order/Delivery/SlipOptionsType.php [hunk 1]
```diff
@@ -26,17 +26,17 @@
...
-use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
 use Symfony\Component\Form\Extension\Core\Type as FormType;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormBuilderInterface;
 
 /**
  * This form class generates the "Options" form in Delivery slips page.
  */
-class SlipOptionsType extends TranslatorAwareType
+class SlipOptionsType extends MultistoreConfigurationType
```
Comment (saulaski) [inline src/PrestaShopBundle/Form/Admin/Sell/Order/Delivery/SlipOptionsType.php:39]: Not sure it is required to extend class from `MultistoreConfigurationType` to apply form extension `MultistoreConfigurationTypeExtension`. Symfony form extension mecanism relies on `FormTypeInterface::getParent()`, which looks correctly implemented here.

**Comment by matks** [comment]
Hi @nenes25 I'm sorry but I think you were too optimistic 😅 . I close this PR as it did not move since November. But if you want to continue, we can always reopen.

**Pull request closed without merging**